templates/CRM/common/civicrm.settings.php.template

```diff
@@ -264,6 +252,22 @@ if (!defined('CIVICRM_UF_BASEURL')) {
   define( 'CIVICRM_UF_BASEURL'      , '%%baseURL%%');
 }
 
+
+/**
+ * Specify a Smarty 3 autoload file.
+ *
+ * If you wish to experiment with Smarty3 you can do so by installing it
+ * to a directory using composer (composer require smarty/smarty:^3)
+ * and then defining the path to that
+ * autoload. file.
+ */
+if (!defined('CIVICRM_SMARTY3_AUTOLOAD_PATH')) {
+  // enable by default for dev & demo sites for now - will soon enable by default for all new installs.
```

Comment:
Based on the doc comment here it looks like smarty3 doesn't work unless you install it yourself using composer, but maybe that's no longer the case?

Reply:
no - it lives in packages now - I can update that

Reply:
@colemanw perhaps the comment @MegaphoneJon made (somewhere else, a while back) about the path being wrong for composer-installed drupal versions was based on the assumption it would be in vendor & packages is fine for D9+?

Reply:
All my tests are failing now because packages isn't under civicrm_root.

Reply:
@demeritcowboy ah - so 2 options

1) we fix the path (if possible)
2) we exclude some drupal variants from the IF

1 is better but I'm not familiar enough with drupal + composer

Reply:
If this is meant to be temporary then 2 is easier. The first might need to have one of those `%%` placeholders populated by the installer?

Reply:
@demeritcowboy let's do 2 for starters & then we can get to 1 later - what is the list of 'opt out' UF?

Reply:
Easiest would just be only turn it on for "Drupal".
I don't know offhand about joomla but opt-out would include Drupal8 and for Standalone it depends how you install it.

Reply:
@demeritcowboy I took a quick look at 1 & from the code it looks like something like this would work - https://github.com/civicrm/civicrm-core/pull/28745 - but I have not tested yet

Reply:
How about https://github.com/civicrm/civicrm-core/pull/28749 ?